src/Raven.Server/Documents/Handlers/Processors/Subscriptions/AbstractSubscriptionsHandlerProcessorForDropSubscriptionConnection.cs

```diff
@@ -1,26 +1,86 @@
         {
+            SubscriptionStorage = subscriptionStorage ?? throw new ArgumentNullException(nameof(subscriptionStorage));
+        }
+
+        private async ValueTask DropSubscriptionAsync(long? subscriptionId, string subscriptionName, string workerId)
+        {
+            using (ServerStore.Engine.ContextPool.AllocateOperationContext(out ClusterOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                var subscription = SubscriptionStorage.GetActiveSubscription(context, subscriptionId, subscriptionName);
+
+                if (subscription != null)
+                {
+                    var result = DropSingleSubscriptionConnection(subscription, string.IsNullOrEmpty(workerId) == false ? workerId : null);
+
+                    if (result == false)
+                    {
+                        HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                        return;
+                    }
+                }
+            }
+
+            await RequestHandler.NoContent();
         }
 
-        protected abstract ValueTask DropSubscriptionAsync(long? subscriptionId, string subscriptionName, string workerId);
+        private bool DropSingleSubscriptionConnection(SubscriptionState subscription, string workerId)
+        {
+            bool result;
+            if (workerId == null)
+            {
+                result = SubscriptionStorage.DropSubscriptionConnections(subscription.SubscriptionId,
+                    new SubscriptionClosedException(
+                        $"Dropped by API request (request ip:{HttpContext.Connection.RemoteIpAddress}, cert:{HttpContext.Connection.ClientCertificate?.Thumbprint})",
+                        canReconnect: false));
+            }
+            else
+            {
+                result = SubscriptionStorage.DropSingleSubscriptionConnection(subscription.SubscriptionId, workerId,
+                    new SubscriptionClosedException(
+                        $"Connection with Id {workerId} dropped by API request (request ip:{HttpContext.Connection.RemoteIpAddress}, cert:{HttpContext.Connection.ClientCertificate?.Thumbprint})",
+                        canReconnect: false));
+            }
+
+            return result;
+        }
 
-        public override async ValueTask ExecuteAsync()
+        protected override RavenCommand<object> CreateCommandForNode(string nodeTag)
         {
-            var subscriptionId = RequestHandler.GetLongQueryString("id", required: false);
-            var subscriptionName = RequestHandler.GetStringQueryString("name", required: false);
-            var workerId = RequestHandler.GetStringQueryString("workerId", required: false);
+            return new DropSubscriptionConnectionCommand(SubscriptionName, SubscriptionId, WorkerId);
```

Comment:
don't you need to use the `nodeTag` here?

Reply:
no we set it as `SelectedNode` here
https://github.com/ravendb/ravendb/blob/499a13ea83904b9673bf99347ba44cf66db5d677/src/Raven.Server/Documents/Handlers/Processors/AbstractHandlerProxyReadProcessor.cs#L35